src/masternode-payments.cpp

```diff
@@ -333,15 +330,21 @@ void CMasternodePayments::ProcessMessage(CNode* pfrom, std::string& strCommand,
 
     } else if (strCommand == NetMsgType::MASTERNODEPAYMENTVOTE) { // Masternode Payments Vote for the Winner
 
+
         CMasternodePaymentVote vote;
         vRecv >> vote;
 
-        if(pfrom->nVersion < GetMinMasternodePaymentsProto()) return;
-
         uint256 nHash = vote.GetHash();
 
         pfrom->setAskFor.erase(nHash);
 
+        // TODO: clear setAskFor for MSG_MASTERNODE_PAYMENT_BLOCK too
+
+        if(pfrom->nVersion < GetMinMasternodePaymentsProto()) return;
+
```

Comment:
Good point, fixed. As a side note, this should probably be moved to the top of `CMasternodePayments::ProcessMessage` to avoid serving outdated nodes completely but that's out of the scope for this PR.